coremasscheck.F90

```diff
@@ -0,0 +1,66 @@
+! Include shortname defintions, so that the F77 code does not have to be modified to
+! reference the CARMA structure.
+#include "carma_globaer.h"
+
+!!  This routine checks if the coremass exceeds the total
+!!
+!! @author Yunqian Zhu
+!! @version Apr-2021
+subroutine coremasscheck(carma, cstate, iz, fixcoremass,logmsg,abort, packagename, rc)
+
+  ! types
+  use carma_precision_mod
+  use carma_enums_mod
+  use carma_constants_mod
+  use carma_types_mod
+  use carmastate_mod
+  use carma_mod
+
+  implicit none
+
+  type(carma_type), intent(inout)      :: carma       !! the carma object
+  type(carmastate_type), intent(inout) :: cstate      !! the carma state object
+  integer, intent(in)                  :: iz
+  logical, intent(in)                  :: fixcoremass !! if we fix the coremass if exceeds total
+  logical, intent(in)                  :: logmsg      !! write message to log file if core mass check fails
+  logical, intent(in)                  :: abort       !! set return code to ERROR and return if  if core mass check fails
+  character(len=*), intent(in)         :: packagename !! string to add to error message
+  integer, intent(inout)               :: rc          !! return code, negative
+
+  ! Local declarations
+  integer                        :: igroup,ibin
+  integer                        :: iepart,i,icore
+  real(kind=f)                   :: coremass
+  real(kind=f)                   :: factor
+
+  real(kind=f), parameter :: thresh  = 1e-10_f     ! relative threshold for core mass and kappa checking
+  real(kind=f), parameter :: thresh1 = 1._f+thresh
+
```

Comment:
If you are going to have thresholds, the constants should be in carma_constants since they are used in other places (e.g. hygroscopicity). Not sure wy you need a threshold nor want one that is greater than one.

Reply:
is done (code change from Yunquian)